alignak/alignakobject.py

```diff
@@ -20,70 +20,159 @@
+        parameters value even if those attributes do not exist previously.
 
+        If debug is set, then this method will print information about the object properties
+
+        The parsing parameter indicates that the object is created from the initial Alignak
+        configuration.
+
+        :param debug: print debug information about the object properties
+        :param params: parameters used to create the object
+        :param parsing: if True, initial creation, else, object unserialization
+        """
         if params is None:
-            return
-        all_props = {}
-        all_props.update(getattr(self, "properties", {}))
-        all_props.update(getattr(self, "running_properties", {}))
+            params = {}
+
+        if debug:
+            print('AlignakObject __init__: %s, %d params' %
+                  (self.__class__, len(params)))
+            print('AlignakObject __init__: %s, params list: %s' %
+                  (self.__class__, [key for key in params]))
+            print('AlignakObject __init__: %s, %d properties' %
+                  (self.__class__, len(self.properties)))
+            print('AlignakObject __init__: %s, properties list: %s' %
+                  (self.__class__, [key for key in self.properties]))
+            print('AlignakObject __init__: %d running properties' %
+                  len(getattr(self, "running_properties", {})))
+            print('AlignakObject __init__: %s, running properties list: %s' %
+                  (self.__class__, [key for key in self.running_properties]))
+
+        # Fresh new object or object update/unserialization?
+        if parsing:
+            self.uuid = uuid.uuid4().hex
+
+        # Create attributes for class properties, running properties and macros (to be confirmed?)
+        object_properties = {}
+        object_properties.update(getattr(self.__class__, "properties", {}))
+        object_properties.update(getattr(self.__class__, "running_properties", {}))
+        # object_properties.update(getattr(self, "macros", {}))
+        index = 0
         for key, value in params.iteritems():
-            if key in all_props and isinstance(all_props[key], SetProp):
+            index = index + 1
+
+            if hasattr(self, key):
+                if debug:
+                    print('AlignakObject __init__: %s still exists (%s)' %
+                          (key, getattr(self, key)))
+                continue
+
+            if key in object_properties and isinstance(object_properties[key], SetProp):
                 setattr(self, key, set(value))
             else:
                 setattr(self, key, value)
 
-        if not hasattr(self, 'uuid'):
+            if debug:
+                print("%c %3d: %s = %s" %
+                      ('!' if key not in object_properties else ' ',
+                       index, key, getattr(self, key, 'not_set')))
+
+        if getattr(self, 'uuid', None) is None:
```

Comment:
Som objects use this function but do not inherit from AlignakObject. Yes, there are !